### DISCUSSION TITLE
topology with tls

### DISCUSSION THREAD

**Discussion by janiu-001** [rabbitmq/messaging-topology-operator#612] [Q&A]
Hi expert, 

I'm using messaging-topology-operator to manage the user, exchange, binding...etc.
And I also configured the tls for the rabbitmq-cluser, according to the docs, we need add SAN when generate secret:
![image](https://github.com/rabbitmq/messaging-topology-operator/assets/19970581/4a959f45-a205-4cd8-8a31-985c3f170713)

In our solution, cert-manger with Let's Encrypt is used to manage the secret.  But it only supports regular DNS domain name (cannot support the format like servicename.mynamespace.svc.cluster.local).

The error is as following with the secret generated by Let's Encrypt
![image](https://github.com/rabbitmq/messaging-topology-operator/assets/19970581/f422ed41-9c62-44d9-8b5a-3adbf5e83eeb)

<img width="1862" alt="image" src="https://github.com/rabbitmq/messaging-topology-operator/assets/19970581/3f9a1656-c57b-47fa-b1f5-ff46674bdec3">

Would we have a way to disable the authentication for messaging-topology-operator? Or do you have any other suggestion?

By the way, it can work well with the secret generated by tl-geen(the tool provided by rabbitmq)

**Comment by janiu-001** [comment]
i tried the env MESSAGING_DOMAIN_NAME

![image](https://github.com/rabbitmq/messaging-topology-operator/assets/19970581/e8998854-c1f4-42af-8ffd-05b5feb836c9)

But looks like it can not resolve the host, will we add some dns confguration ?

![image](https://github.com/rabbitmq/messaging-topology-operator/assets/19970581/185df7b8-9ccd-40b3-a913-ce655a03615e)

**Comment by janiu-001** [comment]
it can work with the following configuration, is it expected solution?

![image](https://github.com/rabbitmq/messaging-topology-operator/assets/19970581/c2bcc1a4-8da8-42d5-8611-f81289d1cb87)

**Reply by ChunyiLyu** [reply]
Yes providing a connection secret is the way to go here. Without setting the secret, topology operator always connects with k8s DNS.
